ui/app/AppLayouts/Wallet/views/collectibles/CollectibleMedia.qml

```diff
@@ -0,0 +1,43 @@
+import QtQuick 2.14
+
+import StatusQ.Components 0.1
+import StatusQ.Core.Theme 0.1
+
+import utils 1.0
+
+StatusRoundedMedia {
+    id: root
+
+    readonly property bool isEmpty: !mediaUrl.toString() && !fallbackImageUrl.toString()
+    property color backgroundColor: Theme.palette.baseColor5
+    property bool isCollectibleLoading: false
+    property string description
+    property string name
+
+    radius: Style.current.radius
+    color: isError || isEmpty ? Theme.palette.baseColor5 : backgroundColor
+
+    Loader {
+        id: loadingCompLoader
+        anchors.fill: parent
+        active: root.isCollectibleLoading || root.isLoading
+        sourceComponent: LoadingComponent {radius: root.radius}
+    }
+
+    Loader {
+        anchors.fill: parent
+        active: (root.isError || root.isEmpty) && !loadingCompLoader.active
+        sourceComponent: LoadingErrorComponent {
+            radius: root.radius
+            text: {
+                if (root.isError && root.componentMediaType === StatusRoundedMedia.MediaType.Unkown) {
+                    return qsTr("Unsupported\nfile format")
+                }
+                if (!root.description && !root.name) {
```

Comment:
hmm empty name and description does not necessarily mean we failed to load collectible metadata.
Also seems kind of useless to pass these two properties for the sole purpose of determining if metadata is present.
Perhaps add some boolean property like `hasSomeMetadata`?

Reply:
I agree with @dlipicar that `description` and `name` are not needed, component api should be as simple as possible and take only what is necessary, and ideally not do any additional logic internally. I also agree with @caybro that every new component should have storybook page with additional controls allowing interaction with the component and checking how particular properties affect look and behavior.